e2e/rdctl.e2e.spec.ts

```diff
@@ -203,13 +229,254 @@ test.describe('HTTP control interface', () => {
+      expect(stderr).toContain('Usage:');
+      expect(stdout).toEqual('');
+    });
+
+    test('api: empty string endpoint should give an error message', async() => {
+      const { stdout, stderr } = await rdctl(['api', '']);
+
+      expect(stderr).toContain('Error: api command: no endpoint specified');
+      expect(stderr).toContain('Usage:');
+      expect(stdout).toEqual('');
+    });
+
+    test('api: complains when more than one arg is given', async() => {
+      const endpoints = ['settings', '/v0/settings'];
+      const { stdout, stderr } = await rdctl(['api', ...endpoints]);
+
+      expect(stderr).toContain(`Error: api command: too many endpoints specified ([${ endpoints.join(' ') }]); exactly one must be specified`);
+      expect(stderr).toContain('Usage:');
+      expect(stdout).toEqual('');
+    });
+
+    test('api: can get the settings using combinations of endpoints and methods', async() => {
+      for (const endpoint of ['settings', '/v0/settings']) {
+        for (const methodSpecs of [[], ['-X', 'GET'], ['--method', 'GET']]) {
+          const { stdout, stderr } = await rdctl(['api', endpoint, ...methodSpecs]);
+
+          if (stderr) {
+            // Do this for looping tests so we know which one failed.
+            console.log(`About to fail for combo ${ methodSpecs.join(' ') } ${ endpoint }`);
+          }
+          expect(stderr).toEqual('');
+          const settings = JSON.parse(stdout);
+
+          expect(['version', 'kubernetes', 'portForwarding', 'images', 'telemetry', 'updater', 'debug', 'pathManagementStrategy']).toMatchObject(Object.keys(settings));
+        }
+      }
+    });
+
+    test('api: set: can read input file from stdin', async() => {
+      if (os.platform() === 'win32') {
+        // do it some other time
+        return;
+      }
+      const settingsFile = path.join(paths.config, 'settings.json');
+      const rdctl = path.join(process.cwd(), 'resources', os.platform(), 'bin', 'rdctl');
+
+      for (const endpoint of ['settings', '/v0/settings']) {
+        for (const methodSpec of ['-X PUT', '--method PUT', '']) {
+          for (const inputSpec of ['--input -', '--input=-']) {
+            const { stdout, stderr } = await bash(`cat ${ settingsFile } | ${ rdctl } api ${ endpoint } ${ methodSpec } ${ inputSpec }`);
+
+            expect(stderr).toBe('');
+            expect(stdout).toContain('no changes necessary');
+          }
+        }
+      }
+      // And complains about a '--input-' flag
+      const { stdout, stderr } = await bash(`cat ${ settingsFile } | ${ rdctl } api settings -X PUT --input-`);
+
+      expect(stdout).toEqual('');
+      expect(stderr).toContain('Error: unknown flag: --input-');
+    });
+
+    test('api: PUT /v0/settings from body', async() => {
+      const settingsFile = path.join(paths.config, 'settings.json');
+      const rdctl = path.join(process.cwd(), 'resources', os.platform(), 'bin', 'rdctl');
```

Comment:
This shadows the `rdctl` function and is causing test failures. (It's also not actually used.)

The same also happens on line 436.

Reply:
Another copy-pasto - it's gone